litestar/contrib/pydantic.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+from pydantic import BaseModel
+
+from litestar.dto.factory.abc import AbstractDTOFactory
+from litestar.dto.factory.field import DTO_FIELD_META_KEY, DTOField
+from litestar.dto.factory.types import FieldDefinition
+from litestar.dto.factory.utils import get_model_type_hints
+from litestar.types.empty import Empty
+from litestar.utils.helpers import get_fully_qualified_class_name
+
+if TYPE_CHECKING:
+    from typing import Any, ClassVar, Collection, Generator
+
+    from pydantic.fields import ModelField
+
+    from litestar.typing import ParsedType
+
+__all__ = ("PydanticDTO",)
+
+DataT = TypeVar("DataT", bound="BaseModel | Collection[BaseModel]")
+
+
+class PydanticDTO(AbstractDTOFactory[DataT], Generic[DataT]):
+    """Support for domain modelling with Pydantic."""
+
+    __slots__ = ()
```

Comment:
If the AbstractDTOFactory specifies slots, no need to reoeat

Reply:
This is required so that the subclass doesn't end up with a `__dict__` attribute and accept arbitrary assignment. E.g.,

```py
from __future__ import annotations


class A:
    __slots__ = ("a",)
    a: int


class B(A):
    __slots__ = ()
    b: int


class C(A):
    c: int


a = A()
a.a = 1

b = B()
b.a = 1
b.b = 2

c = C()
c.a = 1
c.c = 3
```

```
main.py:23: error: Trying to assign name "b" that is not in "__slots__" of type "main.B"  [misc]
Found 1 error in 1 file (checked 1 source file)
```

The only case that mypy and pycharm warn about is `b.b=2`. Without `__slots__` defined on `C`, `c.c` is valid assignment.